app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java

```diff
@@ -174,8 +176,9 @@ public void onClick(final DialogInterface dialog, final int which) {
         }
     }
 
-    private final class EditNameClickListener implements DialogInterface.OnClickListener {
-
+    private static final class EditNameClickListener implements DialogInterface.OnClickListener {
+        private final Activity activity;
+        private final IDatabase database;
```

Comment:
That's true, but we cannot make it a local variable, because the constructor needs the key, and we don't know that key in onCreate yet